kong/plugins/jwt/handler.lua

```diff
@@ -28,6 +28,13 @@ local function retrieve_token(request, conf)
     end
   end
 
+  for _, v in ipairs(conf.cookie_names) do
+    local jwt_cookie = ngx.unescape_uri(ngx.var["cookie_" .. v])
```

Comment:
The access to the `ngx` global in a loop in such a hot code path warrants a caching of the `unescape_uri` function at the top of this module:
```lua
local unescape_uri = ngx.unescape_uri
```

The same way, the `ngx.var` access should be cached locally at the function's level:
```lua
local var = ngx.var

for _, v in ipairs(conf.cookie_names) do
  local escaped_cookie = var["cookie_" .. v]
  if escaped_cookie then
    return unescape_uri(escaped_cookie)
  end
end
```

Reply:
on second looks; the jwt itself is safe-base64 encoded, so the `unescape_uri` call is not required.

Reply:
Yes, was also having second thoughts about it. Curious to hear @mvanholsteijn 's take on why this seemed to be required?

Reply:
I removed the unescape_uri code ; I do not believe it is necessary too.